src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -680,24 +798,120 @@ def get_observation(self, event: Event) -> pd.DataFrame:
             | (df_eligible_dependents["last_year_income"] < 4_300)
         ]
 
-        s_eligible_dependent_id_list = df_eligible_dependents.groupby("guardian_id").apply(
-            lambda df_g: list(df_g.index)
-        )
+        df = df_eligible_dependents
+        df["dependent_id"] = df.index
+        df["tax_year"] = event.time.year - 1
 
-        s_eligible_dependent_address_id_list = df_eligible_dependents.groupby(
-            "guardian_id"
-        ).apply(lambda df_g: list(df_g["address_id"]))
+        # merge in simulant columns based on simulant id
+        for col in [
+            "first_name_id",
+            "middle_name_id",
+            "last_name_id",
+            "age",
+            "date_of_birth",
+            "sex",
+            "ssn",
+            "address_id",
+            "housing_type",
+        ]:
+            df[col] = df["dependent_id"].map(pop_full[col])
```

Comment:
What is the situation in which these columns are not already present? They seem like they all should be.

Reply:
> What is the situation in which these columns are not already present? They seem like they all should be.

Huh, good point.  I didn't think that through!